### PR TITLE
Cache query string params

### LINKED ISSUE #124: Saved or cached schedule
Good Evening to the great team behind ANU Timetable :D
I think it would be better if the courses that I have selected is saved so that the next time I access the site I don't have to refill my courses again. I saw that you currently the web works by saving the courses on the HTML parameters, which is temporary. Therefore, if this sounds like too much of work, then just skip it. The timetable is already great!

Label: Feature request

_Originally submitted by anonymous via the Microsoft Form_

### PR DESCRIPTION
**Pull request by CraftyDH**
Fixes #124 

This is done by saving param changes to localstorage object "savedQueryParams". This is then read and applied when a user has a saved state and visits the site without any url parameters.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -59,6 +59,8 @@ let App = () => {
   )
   useEffect(() => localStorage.timeZone = timeZone, [timeZone])
 
+  findPreviousState()
```
Comment (pl4nty) [inline src/App.js:62]: Maybe loadCachedQSIfNotExists()? Bit verbose ik, honestly probs easier to just put the whole function here

**Comment by pl4nty** [comment]
Welp I missed this note in the CI docs

> Workflows triggered by pull_request_target events are run in the context of the base branch. Since the base branch is considered trusted, workflows triggered by these events will always run, regardless of approval settings.

You'll have to push another commit to trigger the staging env, sorry!

**Review by pl4nty — APPROVED** [review]
LGTM, can merge when it passes CI. I'm planning a state rearch in future, but that's a while away so this will be great for now.

File: src/App.js (hunk 1 above)
Comment (CraftyDH) [reply src/App.js:62]: I sort of wanted to keep the URL functionality close to each other, but I'll change the function name.

**Pull request merged**